DXMainClient/DXMainClient.csproj

```diff
@@ -1,660 +1,85 @@
-    <ErrorReport>prompt</ErrorReport>
-    <Optimize>true</Optimize>
-    <Prefer32Bit>false</Prefer32Bit>
+  <PropertyGroup Condition="'$(Configuration)' == 'TSDXDebug' Or '$(Configuration)' == 'TSGLDebug' Or '$(Configuration)' == 'TSXNADebug' Or '$(Configuration)' == 'TSDXRelease' Or '$(Configuration)' == 'TSGLRelease' Or '$(Configuration)' == 'TSXNARelease'">
+    <DefineConstants>$(DefineConstants);TS</DefineConstants>
   </PropertyGroup>
-  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'AresRelease|XNAFramework' ">
-    <OutputPath>bin\XNAFramework\Release\</OutputPath>
-    <DefineConstants>TRACE;XNA ARES</DefineConstants>
-    <Optimize>true</Optimize>
-    <DebugType>pdbonly</DebugType>
-    <PlatformTarget>x86</PlatformTarget>
-    <ErrorReport>prompt</ErrorReport>
-    <TargetFrameworkVersion>v4.0</TargetFrameworkVersion>
-    <WarningLevel>4</WarningLevel>
+  <PropertyGroup Condition="'$(Configuration)' == 'AresDXDebug' Or '$(Configuration)' == 'AresDXRelease' Or '$(Configuration)' == 'TSDXDebug' Or '$(Configuration)' == 'TSDXRelease' Or '$(Configuration)' == 'YRDXDebug' Or '$(Configuration)' == 'YRDXRelease'">
+    <DefineConstants>$(DefineConstants);WINDOWSDX</DefineConstants>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'AresDebug|SharpDX'">
-    <DebugSymbols>true</DebugSymbols>
-    <OutputPath>bin\SharpDX\Debug\</OutputPath>
-    <DefineConstants>TRACE;DEBUG;ARES</DefineConstants>
-    <DebugType>full</DebugType>
-    <PlatformTarget>AnyCPU</PlatformTarget>
-    <LangVersion>7.3</LangVersion>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
-    <Prefer32Bit>false</Prefer32Bit>
+  <PropertyGroup Condition="'$(Configuration)' == 'AresGLDebug' Or '$(Configuration)' == 'AresGLRelease' Or '$(Configuration)' == 'TSGLDebug' Or '$(Configuration)' == 'TSGLRelease' Or '$(Configuration)' == 'YRGLDebug' Or '$(Configuration)' == 'YRGLRelease'">
+    <DefineConstants>$(DefineConstants);WINDOWSGL</DefineConstants>
   </PropertyGroup>
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'AresDebug|WindowsGL'">
-    <DebugSymbols>true</DebugSymbols>
-    <OutputPath>bin\WindowsGL\Debug\</OutputPath>
-    <DefineConstants>TRACE;DEBUG;WINDOWSGL ARES</DefineConstants>
-    <DebugType>full</DebugType>
-    <PlatformTarget>AnyCPU</PlatformTarget>
-    <LangVersion>7.3</LangVersion>
-    <ErrorReport>prompt</ErrorReport>
-    <CodeAnalysisRuleSet>MinimumRecommendedRules.ruleset</CodeAnalysisRuleSet>
-    <Prefer32Bit>false</Prefer32Bit>
+  <PropertyGroup Condition="'$(Configuration)' == 'AresXNADebug' Or '$(Configuration)' == 'AresXNARelease' Or '$(Configuration)' == 'TSXNADebug' Or '$(Configuration)' == 'TSXNARelease' Or '$(Configuration)' == 'YRXNADebug' Or '$(Configuration)' == 'YRXNARelease'">
+    <DefineConstants>$(DefineConstants);XNA</DefineConstants>
   </PropertyGroup>
```

Comment:
You may used custom property in msbuild project.

like this
```xml
<PropertyGroup>
  <Engine>DX</Engine><!-- DX, OGL, XNA -->
  <Game>Ares</Game><!-- Ares, TS, YR -->
  <Configurations>Debug;Release</Configurations><!-- Make it default -->
</PropertyGroup>

<PropertyGroup>
  <!-- Engines -->
  <DefineConstants Condition="'$(Engine)' == 'DX'">$(DefineConstants);WINDOWSDX</DefineConstants>
  <DefineConstants Condition="'$(Engine)' == 'OGL'">$(DefineConstants);WINDOWSGL</DefineConstants>
  <DefineConstants Condition="'$(Engine)' == 'XNA'">$(DefineConstants);XNA</DefineConstants>

  <!-- Games-->
  <DefineConstants Condition="'$(Game)' == 'Ares'">$(DefineConstants);Ares</DefineConstants>
  <DefineConstants Condition="'$(Game)' == 'TS'">$(DefineConstants);TS</DefineConstants>
  <DefineConstants Condition="'$(Game)' == 'YR'">$(DefineConstants);YR</DefineConstants>
</PropertyGroup>
```

And you can build it by command-line like this.
```cmd
dotnet build -c:Release -p:Engine=DX -p:Game=Ares
```

This will keep the project files clean.

[MSBuild](https://docs.microsoft.com/visualstudio/msbuild/) is really powerful!